### PR TITLE
fix(auth): guard OAuth :strategy route — 404 unknown/disabled + stateless authenticate

### LINKED ISSUE #3900: 🔒 fix(auth): guard OAuth :strategy route — 404 unknown/disabled provider + stateless authenticate
## Problem
`GET /api/auth/:strategy` (`modules/auth/routes/auth.routes.js`) is a greedy catch-all. `oauthCall` (`modules/auth/controllers/auth.controller.js`) calls `passport.authenticate(req.params.strategy)` with no validation, no callback, no `{ session: false }`. Any unmatched path segment becomes a passport strategy name:
- unknown strategy (e.g. `/api/auth/me`, `/api/auth/callback`) → passport throws synchronously `Unknown authentication strategy` → 500
- a provider in `ALLOWED_PROVIDERS` but not enabled/registered → same 500
- an enabled OAuth provider → `passport.authenticate('google')` defaults to `session: true` → `Login sessions require session support. Did you forget to use express-session middleware?` (the stack is stateless JWT, no express-session)

`oauthCallback` has the same unguarded `req.params.strategy` → the synchronous throw fires before its error callback. `ALLOWED_PROVIDERS` already exists in the controller but is used only in the profile resolver, not the route handlers.

## Fix
- Guard `oauthCall` + `oauthCallback`: reject when strategy is not in `ALLOWED_PROVIDERS` OR the provider is not enabled in config → return 404 (not 500).
- Pass `{ session: false }` to `passport.authenticate` (stateless JWT).
- Add tests for `oauthCall` (none today): unknown → 404, disabled → 404, enabled+allowed → delegates.

## Why
Probe traffic to `/api/auth/*` produces 500s + error-tracking noise; enabling an OAuth provider would break on the session error. Hardening turns these into clean 404s and makes OAuth enable-able.

Created via /dev:issue

### PR DESCRIPTION
**Pull request by PierreBrisorgueil**
## Summary

- What changed: `GET /api/auth/:strategy` and its `/callback` route no longer pass `req.params.strategy` straight into `passport.authenticate()`. A new `isEnabledOAuthProvider()` guard rejects any strategy name that is not both allowlisted (`ALLOWED_PROVIDERS`) and actually registered with passport, returning a clean `404` (`OAUTH_PROVIDER_NOT_FOUND`) instead of a synchronous throw. `oauthCall` now also passes `{ session: false }` to `passport.authenticate()` (stateless JWT stack, no `express-session` middleware mounted).
- Why: any unmatched path segment under `/api/auth/*` was reaching `passport.authenticate(strategy)` unguarded. An unknown name (e.g. probe traffic hitting `/api/auth/me`) made passport throw synchronously ("Unknown authentication strategy") → 500. A provider that's allowlisted but not registered (missing credentials) hit the same throw. An actually-registered provider defaulted to `session: true` and would fail immediately with "Login sessions require session support" on a stack with no session middleware. This closes all three off as clean, testable 404s / correct stateless auth.
- Related issues: Closes #3900

## Scope

- Module(s) impacted: `modules/auth` (controller only — no route/schema changes)
- Cross-module impact: `none` — the one line touched outside `modules/auth` is a pre-existing `modules/invitations` integration test assertion that exercises the same `/api/auth/google` route; it now asserts the new deliberate 404 contract instead of "any non-404", no `invitations` module logic changed.
- Risk level: `low`

## Validation

- [x] `npm run lint`
- [x] `npm test`
- [ ] Manual checks done (if applicable)

## Guardrails check

- [x] No secrets or credentials introduced (`.env*`, `secrets/**`, keys, tokens)
- [x] No risky rename/move of core stack paths
- [x] Changes remain merge-friendly for downstream projects
- [x] Tests added or updated when behavior changed

## Reviewer notes

- The guard uses `passport._strategy(strategy)` — this is `@api private` in passport, but it is the only source-of-truth for "is this strategy registered" in passport@0.7.0 (no public accessor exists; only `use()`/`unuse()`). A config-based check would duplicate each provider's own credential-presence gating (`strategies/local/{google,apple}.js`) and risk drift. Deliberate trade-off, documented inline on `isEnabledOAuthProvider()`.
- `{ session: false }` is passed to `passport.authenticate` in `oauthCall` (the default `session: true` was the root cause of the "Login sessions require session support" error on this stateless-JWT stack). It is intentionally **not** added to `oauthCallback` — passport's callback form (`passport.authenticate(strategy, callback)`) never calls `req.logIn()` / reads `options.session`, so it would be a dead option there. Both are commented inline for the next reader.
- A 4-lens adversarial pre-push review already ran on this diff (security + correctness clean). It found 2 test-coverage gaps, both fixed before this PR was opened: `oauthCallback`'s 404-guard branch is now covered by the new unit test file, and a pre-existing session-mint regression test (identity-trust logic) had its coverage restored past the new guard so it still reaches the code it was written to protect.

## Notes for reviewers

- Security considerations: this closes an unauthenticated-request path that could throw synchronously past intended error handling (500 on probe traffic); no new attack surface is introduced — the guard is strictly more restrictive than the previous unguarded call.
- Mergeability considerations: none — isolated to `modules/auth`, no shared `lib/`/`config/` changes.
- Follow-up tasks (optional): none.

## Summary by CodeRabbit

* **Bug Fixes**
  * OAuth sign-in and callback routes now reject unsupported or unregistered providers with a clear 404 error.
  * OAuth authentication is now handled more safely in stateless mode, reducing the chance of unwanted session behavior.
  * Existing login flows continue to redirect properly on authentication failures, with improved route handling for legacy paths.

* **Tests**
  * Expanded coverage for OAuth provider validation and callback behavior.
  * Strengthened integration checks to prevent mock leakage and verify the correct error response.

### CODE REVIEW

File: modules/auth/controllers/auth.controller.js [hunk 1]
```diff
@@ -439,6 +439,31 @@ const token = async (req, res) => {
...
+ * authentication strategy") for an unregistered name.
+ * `passport._strategy` is documented `@api private` in passport's own source, but it
+ * is the single source of truth for "is this strategy registered" — asking config
+ * directly would mean duplicating each provider's own credential-presence check here
+ * and risking drift from modules/auth/strategies/local/*.js over time.
+ * @param {string} strategy - strategy name from req.params.strategy
+ * @returns {boolean} true when strategy is both allowlisted and registered with passport
+ */
+const isEnabledOAuthProvider = (strategy) => ALLOWED_PROVIDERS.has(strategy) && Boolean(passport._strategy(strategy));
+
```
[the remark this answers is not in the document]
Comment (PierreBrisorgueil) [reply modules/auth/controllers/auth.controller.js:466]: Deliberate, documented trade-off. passport@0.7.0 exposes **no public API** to query whether a strategy is registered (only `use()`/`unuse()`) — `_strategy` is the single source of truth for "is this name registered", which is exactly the throw (`Unknown authentication strategy`) this guard prevents. A config-based check would duplicate each provider's own credential-presence gating in `strategies/local/{google,apple}.js` and drift from it over time. passport is pinned `^0.7.0`, so a major bump that removed `_strategy` would be a deliberate, reviewed upgrade (and would fail loudly in the unit tests that mock it). Keeping the private-API call with its explanatory comment. Resolving as documented won't-change.
